R/spatialCoords.R

```diff
@@ -0,0 +1,76 @@
+
+# spatialCoords ----------------------------------------------------------------
+
+#' @rdname SpatialExperiment-methods
+#' @importFrom SingleCellExperiment int_colData<-
+#' @export
+setMethod("spatialCoords", 
+    "SpatialExperiment",
+    function(x, withDimnames=TRUE) {
+        out <- int_colData(x)$spatialCoords
+        if (withDimnames) 
+            rownames(out) <- colnames(x)
+        return(out)
+    })
+
+#' @rdname SpatialExperiment-methods
+#' @importFrom SingleCellExperiment int_colData<-
+#' @export
+setReplaceMethod("spatialCoords", 
+    c("SpatialExperiment", "matrix"),
```

Comment:
there is a warning that I suppose is coming from the "matrix" instead of a "value"

```
Warning:     ‘spatialCoords<-’
    ‘\S4method{spatialCoords<-}{SpatialExperiment,NULL}’
    ‘\S4method{spatialCoords<-}{SpatialExperiment,matrix}’
  The argument of a replacement function which corresponds to the right
  hand side must be named ‘value’.
```

Reply:
Yeah I saw. But I don’t understand because it’s good locally. I’m on it…

Reply:
indeed I don't get why, but have you tried something like: `c("SpatialExperiment", "value")` ?

Reply:
I think that's because the generic is defined as `setGeneric("spatialCoords<-", function(x, value, withDimnames=TRUE) standardGeneric("spatialCoords<-"))`

Reply:
…the signature has to be classes, so “missing” and “matrix” (not “value”) is correct. It’s saying the generic and methods don’t match. But I don’t see why not.

Reply:
yes, I saw that, I'm looking over the internet, but still I'm not able to understand the motivation for this warning.